update product-level-security-guidelines.md (#2365)
wso2/docs-is: en/docs/administer/product-level-security-guidelines.md

If mutual SSL is enabled, [enable intermediate certificate validation](../../develop/authenticating-and-authorizing-rest-apis/#configure-intermediate-certificate-validation) as well to make sure that only certificates signed by the issuers mentioned in the **IntermediateCertValidation** configuration are allowed to be used during mutual SSL authentication.

If mutual SSL authentication capabilities are not required, you can [disable it](../../administer/enabling-mutual-ssl/#enabling-mutual-ssl-in-the-wso2-is).

> ## Configuring Application Authentication

Reviewer: ```suggestion
## Configuring application authentication
```

> Client authentication is used to identify the application or the client that is making the request. There are default credentials used by the out-of-the-box web applications to authenticate with Product's REST APIs that are marked as secure under the 'ResourceAccessControl' tag of identity.xml file. 

Reviewer: ```suggestion
Client authentication is used to identify the application or the client that is making the request. The web applications provided out-of-the-box use a set of default credentials to authenticate with WSO2 Identity Server REST APIs that are marked as **secure** under the 'ResourceAccessControl' tag of the `identity.xml` file. 
```

> Client authentication is used to identify the application or the client that is making the request. There are default credentials used by the out-of-the-box web applications to authenticate with Product's REST APIs that are marked as secure under the 'ResourceAccessControl' tag of identity.xml file. 

Reviewer 2: ```suggestion
Client authentication is used to identify the application or the client that is making the request. There are default credentials used by the out-of-the-box web applications to authenticate with the WSO2 Identity Server's REST APIs that are marked as **secure** under the 'ResourceAccessControl' tag of the`<IS_HOME>/repository/conf/identity/identity.xml` file. 
```

> In order to change the default credentials do the below steps:

Reviewer: ```suggestion
Follow the steps below to change the default credentials. 
```

> In order to change the default credentials do the below steps:

Reviewer 2: ```suggestion
In order to change the default credentials, follow the steps given below. 
```

> 1.  Before apply the configurations, make sure that you get the latest WUM updates for this release. You can use [WSO2 

Reviewer 2: ```suggestion
1.  Before applying the configurations, make sure that you get the latest WUM updates for this release. See [WSO2 
```

Update Manager](https://wso2.com/updates/wum)(WUM) to get any
    fixes or latest updates for this release.

    !!! note "Important"
        If you are upgrading to use this version in your production
        environment, use WSO2 Update Manager to get the latest updates
        available for WSO2 IS 5.9.0. For more information on how to use WSO2
        Update Manager, see [Updating WSO2 Products](https://docs.wso2.com/display/updates/Using+WSO2+Update+Manager).        

> 2.  Shutdown the server, if you have already started. 

Reviewer 2: ```suggestion
2.  Shut the server down in case you have already started it. 
```

> 3.  Add the following configuration changes to /repository/conf/deployment.toml file.

Reviewer: ```suggestion
3.  Add the following configuration changes to the `<IS_HOME>/repository/conf/deployment.toml` file.
```

> 3.  Add the following configuration changes to /repository/conf/deployment.toml file.

Reviewer 2: ```suggestion
3.  Add the following configuration changes to the ` /repository/conf/deployment.toml` file.
```

    
    - Add the 'app_password' property value of '[identity.auth_framework.endpoint]' to a preferred password.
      
       ``` toml
       [identity.auth_framework.endpoint] 
       app_password="<value of preferred password>"

        ```  
    - Add SHA-256 hash value of the added 'app_password' property value to the 'hash' property value of '[account_recovery.endpoint.auth]'.
    
         ``` toml
         [account_recovery.endpoint.auth]
         hash="<SHA-256 hash of the newly added app_password property value>"

         ```  

>     If the 'authenticationendpoint' web app is hosted externally. Open the 'EndpointConfig.properties' file found in the root of the 'authenticationendpoint' folder. Change the 'app.password' property value to the value added as 'app_password' in the 'deployment.toml' file. 

Reviewer 2: ```suggestion
    If the 'authenticationendpoint' web app is hosted externally, open the 'EndpointConfig.properties' file found in the root of the 'authenticationendpoint' folder. Change the 'app.password' property value to the value added as 'app_password' in the 'deployment.toml' file. 
```

>     And do the same changes to the 'EndpointConfig.properties' file located in the "/WEB-INF/classes'' directory.

Reviewer 2: ```suggestion
    Apply the same changes to the 'EndpointConfig.properties' file located in the "/WEB-INF/classes'' directory.
```

>     If the 'accountrecoveryendpoint' web app is hosted externally. Open the 'RecoveryEndpointConfig. properties' file found in the root of the 'accountrecoveryendpoint' folder. Change the 'app.password' property value to the value added as 'app_password' in the 'deployment.toml' file. 

Reviewer 2: ```suggestion
    If the 'accountrecoveryendpoint' web app is hosted externally, open the 'RecoveryEndpointConfig. properties' file found in the root of the 'accountrecoveryendpoint' folder. Change the 'app.password' property value to the value added as 'app_password' in the 'deployment.toml' file. 
```

    And do the same changes to the “RecoveryEndpointConfig.properties” file located in the "/WEB-INF/classes'' directory.

> 4.  Once above configuration changes are completed, restart the server with :

Reviewer 2: ```suggestion
4.  Once these changes are configured, restart the server with,
```

Follow the steps below to change the default credentials.

1.  Before applying the configurations, make sure that you get the latest WUM updates for this release. See [WSO2 
Update Manager](https://wso2.com/updates/wum)(WUM) to get any
    fixes or latest updates for this release.

    !!! note "Important"
        If you are upgrading to use this version in your production
        environment, use WSO2 Update Manager to get the latest updates
        available for WSO2 IS 5.9.0. For more information on how to use WSO2
        Update Manager, see [Updating WSO2 Products](https://docs.wso2.com/display/updates/Using+WSO2+Update+Manager).        
2.  Shut the server down in case you have already started it. 
3.  Add the following configuration changes to the `<IS_HOME>/repository/conf/deployment.toml` file.
    
    - Add the `app_password` property and enter a preferred password as the value.
      
       ``` toml
       [identity.auth_framework.endpoint] 
       app_password="<value of preferred password>"

        ```  
    - Add the `hash` property and enter the SHA-256 hash value of the `app_password` as the property value.
    
         ``` toml

>          [account_recovery.endpoint.auth]

Reviewer 3: Are we only explaining the account_recovery.endpoint.auth.hash? what about account_recovery.endpoint.auth.name?

Author: We can omit it as there is no security requirement for changing the name.

If mutual SSL is enabled, [enable intermediate certificate validation](../../develop/authenticating-and-authorizing-rest-apis/#configure-intermediate-certificate-validation) as well to make sure that only certificates signed by the issuers mentioned in the **IntermediateCertValidation** configuration are allowed to be used during mutual SSL authentication.

If mutual SSL authentication capabilities are not required, you can [disable it](../../administer/enabling-mutual-ssl/#enabling-mutual-ssl-in-the-wso2-is).

> ## Configuring application authentication

Reviewer 4: ```suggestion
## Configuring client authentication
```
